doc/concepts/data-concepts/provenance.md

```diff
@@ -0,0 +1,67 @@
+# Provenance
+
+Provenance enables Pachyderm users to go back in time and see the state of
```

Comment:
This paragraph is awesome! This is really well-captured. A few tweaks to improve the accuracy slightly:

Data Versioning enables Pachyderm users to go back in time and see the state of a
data set (repo) at a particular moment in time.  Data Provenance (aka: Data Lineage) (if we want to be pretentious we can say "from the french word _provenir_ which means _the place of origin_ "-- since provenance isn't a word everyone actually knows)  tracks the
dependencies and relationships _between_ datasets.

Provenance answers not only the question of where the data comes from,
but also how the data was transformed along the way. Data scientists use provenance in
root cause analysis to improve code, workflows, and their understanding of the data and its implications on final results. Data scientists need to have confidence in the information with which they operate. They need
to be able to reproduce the results and sometimes go through the whole
data transformation process from scratch multiple times, which makes data
provenance one of the most critical aspects of data analysis. If your
computations result in unexpected numbers, the first place to look
is the historical data and code that gives insights into possible flaws in the
transformation chain or the data itself.